Source/Core/Common/MemArena.h

```diff
@@ -15,18 +15,76 @@ namespace Common
 {
 // This class lets you create a block of anonymous RAM, and then arbitrarily map views into it.
 // Multiple views can mirror the same section of the block, which makes it very convenient for
-// emulating
-// memory mirrors.
-class MemArena
+// emulating memory mirrors.
+class MemArena final
 {
 public:
+  MemArena();
+  ~MemArena();
+  MemArena(const MemArena&) = delete;
+  MemArena(MemArena&&) = delete;
+  MemArena& operator=(const MemArena&) = delete;
+  MemArena& operator=(MemArena&&) = delete;
+
+  // Allocate the singular memory segment handled by this MemArena. This will be the actual
+  // 'physical' available memory for this arena. After allocation, it can be interacted with using
+  // CreateView() and ReleaseView(). This is used to make a mappable region for emulated memory.
+  //
+  // Parameters:
+  //  - 'size' is the amount of bytes that should be allocated in this region.
```

Comment:
You may want to turn these comments into doxygen style comments with `///` or `/**` and use `@param` here

clang has a -Wdocumentation warning level that can flag anything if the parameter names change in the future through refactoring and the comment falls out of sync

Reply:
I matched it to the format used in `BitUtils.h`, I assume that's good?